File: engine/model/src/main/scala/tools/packet/showitemgaininchat/ShowOwnPetLevelUp.scala
package tools.packet.showitemgaininchat

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ShowOwnPetLevelUp(private var _index: Byte) extends PacketInput {
  def index: Byte = _index

  override def opcode(): SendOpcode = SendOpcode.SHOW_ITEM_GAIN_INCHAT
}

File: engine/engine-model/src/main/java/tools/packet/showitemgaininchat/ShowOwnPetLevelUp.java
package tools.packet.showitemgaininchat;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ShowOwnPetLevelUp(Byte index) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.SHOW_ITEM_GAIN_IN_CHAT;
   }
}